test/load-parser-config.test.js

```diff
@@ -100,3 +102,12 @@ test('Throw error if "config" doesn`t exist', async (t) => {
 test('Throw error if "preset" doesn`t exist', async (t) => {
   await t.throwsAsync(loadParserConfig({ preset: "unknown-preset" }, { cwd }), { code: "MODULE_NOT_FOUND" });
 });
+
+test.serial("Load preset and config correctly when importFrom.silent fails", async (t) => {
+  sinon.stub(importFrom, "silent").returns(undefined);
```

Comment:
i dont think sinon will work as expected here since this is now an esm project. i would bet that `undefined` is not actually being returned from silent when stubbed this way

Reply:
we've been using testdouble.js in other parts of the project for similar goals

Reply:
It actually does work! The reason I added this test was that test coverage went down because the `silent()` method would always load the presets correctly, so the part after it was never called. Since that got test coverage back to 100%, it proves the sub works as expected

Reply:
> we've been using testdouble.js in other parts of the project for similar goals

Yes I noticed that in `release-notes-generator` after implementing it here with `sinon`, so I tried to use testdouble for the stub there, but couldn't get it to work after a few hours of trial (can't remember why) and resigned myself to adding `sinon` as a dev dependency on `release-notes-generator` as well ..

Reply:
hmm, interesting. my understanding was that sinon still wasnt able to stub actual imports in an esm context because they are not overridable without a loader. i believe what you are saying, but i dont understand how it would work because it _shouldnt_ based on what i understand. 🤔  this is why we moved previous use of sinon over to testdouble in other areas.

Reply:
now that you're saying it, I do understand the same thing :thinking: just ran some checks to make sure it wasn't due to some shenanigans, and indeed `sinon` is able to correctly stub the function export from `import-from-esm`!

now as to answer the "why?", I'm afraid I can't do that :shrug: 

![image](https://github.com/semantic-release/commit-analyzer/assets/11234273/4571f3a9-3e25-4d13-8ccd-870190e247af)
---
![image](https://github.com/semantic-release/commit-analyzer/assets/11234273/39f39003-c955-437f-86f9-c072300756f9)

Reply:
fair enough 🤷🏼 

we can figure out why later. as long as we can get these up to date with the mainline and everything passing, i'm good to merge in that way